src/runtime/local/kernels/Transpose.h

```diff
@@ -87,36 +87,53 @@ struct Transpose<DenseMatrix<VT>, DenseMatrix<VT>> {
 template<typename VT>
 struct Transpose<CSRMatrix<VT>, CSRMatrix<VT>> {
     static void apply(CSRMatrix<VT> *& res, const CSRMatrix<VT> * arg, DCTX(ctx)) {
-        const size_t numRows = arg->getNumRows();
-        const size_t numCols = arg->getNumCols();
-        
-        if(res == nullptr)
-            res = DataObjectFactory::create<CSRMatrix<VT>>(numCols, numRows, arg->getNumNonZeros(), false);
-        
-        const VT * valuesArg = arg->getValues();
-        const size_t * colIdxsArg = arg->getColIdxs();
-        const size_t * rowOffsetsArg = arg->getRowOffsets();
-        
-        VT * valuesRes = res->getValues();
-        VT * const valuesResInit = valuesRes;
-        size_t * colIdxsRes = res->getColIdxs();
-        size_t * rowOffsetsRes = res->getRowOffsets();
-        
-        auto* curRowOffsets = new size_t[numRows + 1];
-        memcpy(curRowOffsets, rowOffsetsArg, (numRows + 1) * sizeof(size_t));
-        
-        rowOffsetsRes[0] = 0;
-        for(size_t c = 0; c < numCols; c++) {
-            for(size_t r = 0; r < numRows; r++)
-                if(curRowOffsets[r] < rowOffsetsArg[r + 1] && colIdxsArg[curRowOffsets[r]] == c) {
-                    *valuesRes++ = valuesArg[curRowOffsets[r]];
-                    *colIdxsRes++ = r;
-                    curRowOffsets[r]++;
-                }
-            rowOffsetsRes[c + 1] = valuesRes - valuesResInit;
+      const size_t numRows = arg->getNumRows();
+      const size_t numCols = arg->getNumCols();
+
+      if(res == nullptr)
+        res = DataObjectFactory::create<CSRMatrix<VT>>(numCols, numRows, arg->getNumNonZeros(), false);
+
+      const VT * valuesArg = arg->getValues();
+      const size_t * colIdxsArg = arg->getColIdxs();
+      const size_t * rowOffsetsArg = arg->getRowOffsets();
+
+      const size_t numNonZeros = arg->getNumNonZeros();
+
+      VT * valuesRes = res->getValues();
+      size_t * colIdxsRes = res->getColIdxs();
+      size_t * rowOffsetsRes = res->getRowOffsets();
+
+      // Maybe use memset instead?
+      // memset(rowOffsetsRes, 0, sizeof(size_t) * (numCols + 1));
```

Comment:
Influenced by [1] and as it's also used in the originating code, I vote for std::fill. 

[1] https://stackoverflow.com/questions/77500815/practical-advantages-of-stdcopy-stdequal-stdfill-over-memcpy-memset-me

Reply:
Addressed in [f56d582](https://github.com/daphne-eu/daphne/pull/798/commits/f56d582ccd5282a536751dac8ca09b987b4442e5)